core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -1502,14 +1504,20 @@
 
 (defn parse
   "Responds to a msg-parse message that creates a prepared-statement."
-  [{:keys [conn-state]} {:keys [query]}]
+  [{:keys [conn-state]} {:keys [metrics-registry]} {:keys [query]}]
 
   (let [{:keys [session watermark-tx-id]} @conn-state
         {:keys [^Clock clock], session-parameters :parameters} session]
 
-    (interpret-sql query {:default-tz (.getZone clock)
-                          :watermark-tx-id watermark-tx-id
-                          :session-parameters session-parameters})))
+    (try
+      (interpret-sql query {:default-tz (.getZone clock)
+                            :watermark-tx-id watermark-tx-id
+                            :session-parameters session-parameters})
+      (catch Exception e
+        (when metrics-registry
```

Comment:
You probably don't want to be adding the counter every time here - likely not the most costly of things but we tend to get in the habit of creating as much as we might need ahead of time rather than at runtime.

I'd probably stick the Counter object next to conn-state (i.e. on server, IIRC) so that you don't have to pass a separate opts thing all the way down too :slightly_smiling_face:

Reply:
Ah, you mean add both the counter and the metrics-registry to conn-state?

Reply:
should just need the counter, I don't think you need the registry after you've used it to add the counter?

Reply:
Yup, not thinking that one through; thanks James